docs/upgrading-to-integration.asciidoc

```diff
@@ -0,0 +1,169 @@
+[[upgrade-to-apm-integration]]
+=== Upgrade to the Elastic APM integration
+
+* <<why-upgrade-to-integrations>>
+* <<apm-arch-upgrade>>
+* <<apm-integration-upgrade-limitations>>
+* <<apm-integration-upgrade-steps>>
+* <<apm-integration-upgrade-steps-ess>>
+
+[discrete]
+[[why-upgrade-to-integrations]]
+=== Why upgrade?
+
+The new APM ecosystem offers a number of benefits:
+
+**{fleet}**:
+
+* A single, unified way to add monitoring for logs, metrics, traces, and other types of data to each host -- install one thing instead of multiple
+* Central, unified configuration management -- no need to edit multiple configuration files
+
+**Data streams**:
+
+* Reduced number of fields per index, better space efficiency, and faster queries
+* More granular data control
+* Errors and metrics data streams are shared with other data sources -- which means better long-term integration with the logs and metrics apps
+* Removes template inheritance for ILM policies and makes use of new {es} index and component templates
+* Fixes <<server-resource-exists-not-alias>>
+
+**APM Integration**:
+
+* Easier to install APM on edge machines
+* Improved source map handling and APM agent configuration management
+* Less configuration
+* Easier and less error-prone upgrade path
+
+[discrete]
+[[apm-arch-upgrade]]
+=== APM integration architecture
+
+The Elastic APM integration consists of four components: *APM agents*, the *Elastic APM integration*, *Elasticsearch*, and *Kibana*.
+Generally, there are two ways that these four components can work together:
+
+Apm agents on edge machines send data to a centrally hosted APM integration:
+
+image::./images/apm-architecture.png[Architecture of Elastic APM]
+
+Or, APM agents and the APM integration live on edge machines and enroll via a centrally hosted {agent}:
+
+image::./images/apm-architecture-two.png[Architecture of Elastic APM option two]
+
+NOTE: If RUM is enabled, you must run {agent} centrally.
+If RUM is disabled, {agent} can be run on edge machines.
+
+[discrete]
+[[apm-integration-upgrade-limitations]]
+=== Limitations
+
+There are some limitations to be aware of:
+
+* This change cannot be reverted
+* Currently, only the {es} output is supported
+* APM runs under {agent} which, depending on the installation method, might be require root privileges
+* An {agent} with the APM integration enabled must be managed by {fleet}.
+
+In addition, if you haven't <<upgrade-to-data-streams>> yet, there are additional limitations to be aware of.
+
+[discrete]
+[[apm-integration-upgrade-steps]]
+=== Upgrade a self-managed cluster
```

Comment:
In an effort to not duplicate more content, each step in the upgrade guide links out to docs in either the Fleet User Guide and APM Guide. Is this too confusing?